NeoMathEngine/src/CPU/CpuMathEngineDnnConv.cpp

```diff
@@ -553,319 +536,337 @@ void CCpuMathEngine::BlobConvolution( const CConvolutionDesc& convDesc, const CC
 	}
 }
 
-void CCpuMathEngine::backwardConvolutionAddFilterToOutput( const CCpuConvolutionDesc& desc, const CConstFloatHandle& temp,
-	const CConstFloatHandle* freeTermData, const CFloatHandle& outputData )
-{
-	const float* tempRaw = GetRaw( temp );
-	const float* freeTermDataRaw = freeTermData != nullptr ? GetRaw( *freeTermData ) : nullptr;
-	float* outputDataRaw = GetRaw( outputData );
-
-	const CBlobDesc& input = desc.Result;
-	const CBlobDesc& output = desc.Source;
-	const CBlobDesc& filter = desc.Filter;
-
-	int filterChannels = filter.Depth() * filter.Channels();
-
-	int outputLineStart;
-	int outputLineEnd;
-	OmpGetTaskIndexAndCount( output.ObjectCount() * output.Height(), outputLineStart, outputLineEnd );
-	outputLineEnd += outputLineStart;
-	for( int step = outputLineStart; step < outputLineEnd; ++step ) {
-		float* outputDataPtr = outputDataRaw + step * output.Width() * output.Depth() * output.Channels();
-
-		if( freeTermData != nullptr ) {
-			// Set the free term
-			setVectorToMatrixRows( outputDataPtr, output.Width(), output.Depth() * output.Channels(), freeTermDataRaw );
-		} else {
-			vectorFill( outputDataPtr, 0, output.Width() * output.Depth() * output.Channels() );
-		}
+//------------------------------------------------------------------------------------------------------------
 
-		int batch = step / output.Height();
-		int row = step % output.Height();
-		int inputRowStart = (row + desc.PaddingHeight - filter.Height() + desc.StrideHeight) / desc.StrideHeight;
-		if(inputRowStart < 0) {
-			inputRowStart = 0;
-		}
-		int filterRowBackStart = row - inputRowStart * desc.StrideHeight + desc.PaddingHeight;
-		if(0 > filterRowBackStart || filterRowBackStart >= filter.Height()) {
-			continue;
-		}
-		int filterRowBackEnd = filter.Height() + row - output.Height() - desc.PaddingHeight;
-		if(filterRowBackEnd < 0) {
-			filterRowBackEnd = 0;
-		}
+class CRoundRobinBuffer final {
```

Comment:
You have a bunch of methods `Get*` without comments.

Some of them don't change the state of the object (`GetSourceRowNum`, `GetRaw` and `GetPrevBatchedSourceRowStart`).
Others are also `Get*` but change the state of the object (`GetExistItemsNum`, `GetBufPtr`)

At least add comments to each of them